os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -125,20 +129,53 @@ void pg_timer_int_handler(void *Data)
 			// Switch status back to normal mode after wake up from interrupt
 			pm_activity(PM_IDLE_DOMAIN, 9);
 			break;
+		case PM_PERIODICAL_TIMER:
+			/* Anything else to do ? */
+			break;
 		default:
 			pmdbg("Timer callback triggered without setting timer, Unexpected Error!!!\n");
 			break;
 	}
 	// Reset the global struct
-	g_pm_timer.timer_type = 0;
+	g_pm_timer.timer_type &= ~(timer_type);
 	g_pm_timer.timer_interval = 0;
 }
 
-void up_set_pm_timer(void) {
+void up_set_pm_timer(int pm_timer_type) {
 	// Check whether timer interrupt need to be set
-	if (g_pm_timer.timer_type) {
-		gtimer_init(&g_timer1, TIMER1);
-		gtimer_start_one_shout(&g_timer1, g_pm_timer.timer_interval, (void *)pg_timer_int_handler, NULL);
+	struct timer_notify_s notify;
+	notify.arg   = NULL;
+	notify.pid   = (pid_t)getpid();
+	notify.timer_type = pm_timer_type;
+	int fd = 0;
+	switch (pm_timer_type) {
+		case PM_WAKEUP_TIMER:
+			fd = open("/dev/timer1", O_RDONLY);
+			ioctl(fd, TCIOC_SETMODE, MODE_ONESHOT);
+			break;
+		case PM_LOCK_TIMER:
+			fd = open("/dev/timer2", O_RDONLY);
```

Comment:
Should we assign a specific timer to a specific timer_type Or timer_type should take any timer_i if that is available . Suppose we want to run two periodical timer at the same time, then this code might work.

Reply:
Please view the comment in the previous PR from Mr.Chang: https://github.com/Samsung/TizenRT/pull/6058#discussion_r1492352013. Ultimately, there should be a resource manager for timer, but currently in TizenRT there isn't. Until we have an idea or stable scope being introduced, we can only go with the specific timer usage.